packages/playwright/src/makeTest.ts

```diff
@@ -12,9 +14,80 @@ import {
   trackRun,
   DEFAULT_GLOBAL_RESOURCE_ARCHIVE_TIMEOUT_MS,
 } from '@chromatic-com/shared-e2e';
-import { contentType, takeSnapshot } from './takeSnapshot';
+import { chromaticSnapshots, takeSnapshot } from './takeSnapshot';
 import { createResourceArchive } from './createResourceArchive';
 
+export const performChromaticSnapshot = async (
+  {
+    page,
+    delay,
+    diffIncludeAntiAliasing,
+    diffThreshold,
+    disableAutoSnapshot,
+    forcedColors,
+    pauseAnimationAtEnd,
+    prefersReducedMotion,
+    resourceArchiveTimeout,
+    assetDomains,
+    cropToViewport,
+    shouldTrack = true,
+  }: ChromaticConfig & { page: Page; shouldTrack?: boolean },
+  use: () => Promise<void>,
+  testInfo: TestInfo
+) => {
+  if (shouldTrack) {
+    trackRun();
+  }
+
+  // CDP only works in Chromium, so we only capture archives in Chromium.
+  // We can later snapshot them in different browsers in the cloud.
+  // TODO: I'm not sure if this is the best way to detect the browser version, but
+  // it seems to work
+  if (page.context().browser().browserType().name() !== 'chromium') {
+    await use();
+    return;
+  }
+
+  const completeArchive = await createResourceArchive({
+    page,
+    networkTimeout: resourceArchiveTimeout,
+    assetDomains,
+  });
+  await use();
+
+  if (!disableAutoSnapshot) {
+    await takeSnapshot(page, testInfo);
+  }
+
+  const resourceArchive = await completeArchive();
+  const { testId } = testInfo;
+  const snapshots: Map<string, Buffer> = chromaticSnapshots.get(testId) || new Map();
+
+  const chromaticStorybookParams = {
+    ...(delay && { delay }),
+    ...(diffIncludeAntiAliasing && { diffIncludeAntiAliasing }),
+    ...(diffThreshold && { diffThreshold }),
+    ...(forcedColors && { forcedColors }),
+    ...(pauseAnimationAtEnd && { pauseAnimationAtEnd }),
+    ...(prefersReducedMotion && { prefersReducedMotion }),
+    ...(cropToViewport && { cropToViewport }),
+  };
+
+  await writeTestResult(
+    { ...testInfo, pageUrl: page.url(), viewport: page.viewportSize() },
+    Object.fromEntries(snapshots),
+    resourceArchive,
+    chromaticStorybookParams
+  );
+
+  if (shouldTrack) {
```

Comment:
I don't love adding this `shouldTrack` param to the function just for tests. Is there a reason you didn't mock the `trackComplete` function in the tests instead?

Reply:
Fair point. I've mocked it out now. Thanks for showing me how!